## Group3 - Replication_1
#### Members
1. Andrea Ulloa (20172597)
2. Ana Angulo (20171627)
3. Angela Coapaza (20171636) 


## Question 1:

## An Inferencial Problem: The College-Educated Wage Gap
Using the data from the March Supplement of the U.S. Current Population Survey (2015) , in  this lab, we are going to focus in  payments of the college-educated workers and answer the following inference question:

What is the difference in predicted wages between workers with some college education (scl) versus college graduate workers(clg)?

To investigate the College-Educated Wage Gap, we consider the following log-linear regression model:

\begin{align}
\log(Y) &= \beta'X + \epsilon\\
&= \beta_1 SCL  + \beta_2 CLG + \beta_3'W  + \epsilon,
\end{align}

Where  SCL is the indicator of workers with some college education( 1  if yes 0  otherwise), CLG is the indicator of college graduate workers (1  if yes 0  otherwise) and the  𝑊 's are controls explaining variation in wages. Considering transformed wages by the logarithm, we are analyzing the relative difference in the payment of workers with some college education and college graduate workers. 


## Data Analysis

We consider the same subsample of the U.S. Current Population Survey (2015). Let us load the data set.

In [2]:
load("../../data/wage2015_subsample_inference.Rdata")
attach(data) 

dim(data)

[1] 5150   20

***Variable description***

- occ : occupational classification
- ind : industry classification
- lwage : log hourly wage
- sex : gender (1 female) (0 male)
- shs : some high school
- hsg : High school graduated
- scl : Some College
- clg: College Graduate
- ad: Advanced Degree
- ne: Northeast
- mw: Midwest
- so: South
- we: West
- exp1: experience

In [30]:
colnames(data)  #allows to obtain the name of each variable of the base

[1] "wage"  "lwage" "sex"   "shs"   "hsg"   "scl"   "clg"   "ad"    "mw"   
[10] "so"    "we"    "ne"    "exp1"  "exp2"  "exp3"  "exp4"  "occ"   "occ2" 
[19] "ind"   "ind2"

As we mention before, we focus only on a subset of the data. For this porpouse, we use scl and clg variables.

In [6]:
data <- data[data$scl==1 |data$clg==1,]
dim(data)
data

[1] 3068   20

wage      lwage    sex shs hsg scl clg ad  mw  so  we  ne  exp1 exp2   
10     9.615385 2.263364 1   0   0   0   1   0   0   0   0   1    7.0  0.4900
12    48.076923 3.872802 0   0   0   0   1   0   0   0   0   1   31.0  9.6100
19    28.846154 3.361977 1   0   0   0   1   0   0   0   0   1   22.0  4.8400
30    11.730769 2.462215 1   0   0   0   1   0   0   0   0   1    1.0  0.0100
71    19.230769 2.956512 1   0   0   0   1   0   0   0   0   1    4.0  0.1600
84    12.019231 2.486508 1   0   0   1   0   0   0   0   0   1    5.5  0.3025
89    13.461538 2.599837 1   0   0   1   0   0   0   0   0   1   20.5  4.2025
96    16.346154 2.793993 1   0   0   0   1   0   0   0   0   1   25.0  6.2500
119    8.653846 2.158004 0   0   0   1   0   0   0   0   0   1    3.5  0.1225
135   10.683761 2.368725 1   0   0   1   0   0   0   0   0   1    6.5  0.4225
144   11.538462 2.445686 1   0   0   1   0   0   0   0   0   1    8.5  0.7225
149   19.230769 2.956512 0   0   0   0   1   0   0   0   0   1   11.0  1.2100
152   16.304348 2.791432 1   0   0   1   0   0   0   0   0   1   24.5  6.0025
160   12.000000 2.484907 0   0   0   1   0   0   0   0   0   1    4.5  0.2025
164   16.826923 2.822980 0   0   0   0   1   0   0   0   0   1    8.0  0.6400
176   19.670330 2.979111 0   0   0   0   1   0   0   0   0   1   26.0  6.7600
206   13.942308 2.634928 1   0   0   1   0   0   0   0   0   1   15.5  2.4025
213   10.096154 2.312155 1   0   0   1   0   0   0   0   0   1    9.0  0.8100
225   24.038462 3.179655 1   0   0   0   1   0   0   0   0   1   31.0  9.6100
232   41.208791 3.718652 1   0   0   0   1   0   0   0   0   1   11.0  1.2100
245   21.634615 3.074295 1   0   0   1   0   0   0   0   0   1   17.5  3.0625
264   12.019231 2.486508 1   0   0   1   0   0   0   0   0   1    6.5  0.4225
280    7.211538 1.975682 1   0   0   1   0   0   0   0   0   1   17.5  3.0625
281   11.778846 2.466305 1   0   0   0   1   0   0   0   0   1    7.0  0.4900
286   12.937063 2.560096 0   0   0   1   0   0   0   0   0   1   34.5 11.9025
291   13.076923 2.570849 1   0   0   0   1   0   0   0   0   1   10.0  1.0000
366   21.634615 3.074295 1   0   0   0   1   0   0   0   0   1   23.0  5.2900
368    8.505917 2.140762 0   0   0   1   0   0   0   0   0   1    4.5  0.2025
375   33.989267 3.526045 0   0   0   0   1   0   0   0   0   1   21.0  4.4100
378   15.721154 2.755007 1   0   0   0   1   0   0   0   0   1    2.0  0.0400
...   ...       ...      ... ... ... ... ... ... ... ... ... ... ...  ...    
32313 33.65385  3.516127 0   0   0   1   0   0   0   0   1   0   24.5  6.0025
32314 30.14553  3.406037 1   0   0   0   1   0   0   0   1   0   15.0  2.2500
32347 12.01923  2.486508 1   0   0   1   0   0   0   0   1   0    9.5  0.9025
32362 25.48077  3.237924 0   0   0   0   1   0   0   0   1   0   16.0  2.5600
32388 22.59615  3.117780 1   0   0   1   0   0   0   0   1   0   11.5  1.3225
32416 18.75000  2.931194 1   0   0   1   0   0   0   0   1   0   19.0  3.6100
32419 29.91453  3.398344 1   0   0   0   1   0   0   0   1   0   39.0 15.2100
32458 11.29808  2.424633 1   0   0   0   1   0   0   0   1   0   10.0  1.0000
32468 16.02564  2.774190 1   0   0   1   0   0   0   0   1   0   11.0  1.2100
32493 16.82692  2.822980 1   0   0   0   1   0   0   0   1   0    4.0  0.1600
32497 10.57692  2.358675 0   0   0   0   1   0   0   0   1   0    8.0  0.6400
32498 25.00000  3.218876 0   0   0   0   1   0   0   0   1   0   10.0  1.0000
32502 15.38462  2.733368 0   0   0   1   0   0   0   0   1   0   17.5  3.0625
32504 18.75000  2.931194 1   0   0   0   1   0   0   0   1   0    2.0  0.0400
32509 20.91346  3.040393 1   0   0   1   0   0   0   0   1   0    4.5  0.2025
32517 22.11538  3.096274 0   0   0   1   0   0   0   0   1   0    7.0  0.4900
32528 12.01923  2.486508 0   0   0   1   0   0   0   0   1   0    4.0  0.1600
32539 38.46154  3.649659 1   0   0   1   0   0   0   0   1   0    7.5  0.5625
32540 28.84615  3.361977 0   0   0   0   1   0   0   0   1   0    5.0  0.2500
32551 27.88462  3.328075 0  

In [ ]:
To start our (causal) analysis, we compare the sample means given gender:

In [9]:
install.packages("xtable")
library(xtable)
#we select the objects that we are going to use
Z <- data[which(colnames(data) %in% c("lwage","sex","shs","hsg","scl","clg","ad","ne","mw","so","we","exp1"))]
dim(Z)

Warning message:
"package 'xtable' is in use and will not be installed"


[1] 3068   12

In [10]:
#Then, we filter the database again according to the gender of the individual

data_female <- data[data$sex==1,]     #allows to filter the database
Z_female <- data_female[which(colnames(data) %in% c("lwage","sex","shs","hsg","scl","clg","ad","ne","mw","so","we","exp1"))]


data_male <- data[data$sex==0,]
Z_male <- data_male[which(colnames(data) %in% c("lwage","sex","shs","hsg","scl","clg","ad","ne","mw","so","we","exp1"))]

table <- matrix(0, 12, 3)
table[1:12,1]   <- as.numeric(lapply(Z,mean))
table[1:12,2]   <- as.numeric(lapply(Z_male,mean))
table[1:12,3]   <- as.numeric(lapply(Z_female,mean))
rownames(table) <- c("Log Wage","Sex","Less then High School","High School Graduate","Some College","Gollage Graduate","Advanced Degree", "Northeast","Midwest","South","West","Experience")
colnames(table) <- c("All","Male","Female")
tab<- xtable(table, digits = 4)
tab

All        Male       Female    
Log Wage               3.0000223  3.0384121  2.9569035
Sex                    0.4709909  0.0000000  1.0000000
Less then High School  0.0000000  0.0000000  0.0000000
High School Graduate   0.0000000  0.0000000  0.0000000
Some College           0.4667536  0.4818238  0.4498270
Gollage Graduate       0.5332464  0.5181762  0.5501730
Advanced Degree        0.0000000  0.0000000  0.0000000
Northeast              0.2659713  0.2612446  0.2712803
Midwest                0.2858540  0.2908195  0.2802768
South                  0.2216428  0.2285890  0.2138408
West                   0.2265319  0.2193469  0.2346021
Experience            12.7009452 12.4331485 13.0017301

In particular, the table above shows that the difference in average logwage between men and women for the subset of college-educated workers is equal to  0,0815 . Also, we can notice that other education dimensions are zero since we have restricted the database to people with only college education.

In [15]:
mean(data_male$lwage)-mean(data_female$lwage)

[1] 0.08150856

In [16]:
x=0.08150855508735955 -0.03834473367441493 
x

[1] 0.04316382

Thus, the unconditional gender wage gap is about  8,2 % for the group of never married workers (women get paid less on average in our sample). In comparison to the general data base, the gender-gap is bigger by  4.32 %.

This unconditional (predictive) effect of gender equals the coefficient $\beta$ in the univariate ols regression of $Y$ on $D$:

\begin{align}
\log(Y) &=\beta D + \epsilon.
\end{align}

We verify this by running an OLS regression in R.

In [19]:
install.packages("sandwich")
library(sandwich)

nocontrol.fit <- lm(data$lwage ~ data$sex) #no control model using lwage and the gender dummy
nocontrol.est <- summary(nocontrol.fit)$coef["data$sex",1]  #column of coefficients but only for the gender variable
HCV.coefs <- vcovHC(nocontrol.fit, type = 'HC');
nocontrol.se <- sqrt(diag(HCV.coefs))[2] # Estimated std errors

# print unconditional effect of gender and the corresponding standard error
cat ("The estimated gender coefficient is",nocontrol.est," and the corresponding robust standard error is",nocontrol.se) 


Warning message:
"package 'sandwich' is in use and will not be installed"


The estimated gender coefficient is -0.08150856  and the corresponding robust standard error is 0.01957965

Next, we run an ols regression of $Y$ on $(D,W)$ to control for the effect of covariates summarized in $W$:

\begin{align}
\log(Y) &=\beta_1 D  + \beta_2' W + \epsilon.
\end{align}

Here, we are considering the flexible model from the previous lab. Hence, $W$ controls for experience, education, region, and occupation and industry indicators plus transformations and two-way interactions. 
It's important to notice that we don't include the education controls since, as seen before, the values related to "shs","hsg","ad" are all $0$.

## OlS regression with controls

In [31]:
flex <- lwage ~ sex + (exp1+exp2+exp3+exp4)*(clg+occ2+ind2+mw+so+we)

#   Note that ()*() operation in formula objects in R creates a formula of the sort:
#  (exp1+exp2+exp3+exp4)+ (shs+hsg+scl+clg+occ2+ind2+mw+so+we) +  (exp1+exp2+exp3+exp4)*(shs+hsg+scl+clg+occ2+ind2+mw+so+we)
#  This is not intuitive at all, but that's what it does.

control.fit <- lm(flex, data=data)
control.est <- summary(control.fit)$coef[2,1]

summary(control.fit)




Call:
lm(formula = flex, data = data)

Residuals:
     Min       1Q   Median       3Q      Max 
-1.87897 -0.27894 -0.00777  0.25823  2.85755 

Coefficients: (1 not defined because of singularities)
              Estimate Std. Error t value Pr(>|t|)    
(Intercept)  3.4357559  0.5208785   6.596 5.02e-11 ***
sex         -0.0530623  0.0193532  -2.742 0.006149 ** 
exp1        -0.1962420  0.1996670  -0.983 0.325767    
exp2         4.3105015  2.3797581   1.811 0.070197 .  
exp3        -2.3067253  1.0176306  -2.267 0.023480 *  
exp4         0.3513392  0.1382879   2.541 0.011118 *  
clg          0.2498673  0.1237850   2.019 0.043627 *  
occ22        0.2151993  0.1582208   1.360 0.173900    
occ23        0.0487642  0.2095297   0.233 0.815986    
occ24        0.0281449  0.2300084   0.122 0.902619    
occ25       -0.2711807  0.3944166  -0.688 0.491793    
occ26       -0.2000530  0.2705614  -0.739 0.459725    
occ27       -0.1203371  0.4188017  -0.287 0.773875    
occ28       -0.1719401  0.27214

In [32]:
cat("Coefficient for OLS with controls", control.est)

HCV.coefs <- vcovHC(control.fit, type = 'HC');


Coefficient for OLS with controls -0.05306234

In [34]:
control.se <- sqrt(diag(HCV.coefs))[2] # Estimated std errors
cat("The standard error for OLS with controls", control.se)

The standard error for OLS with controls 0.01835285

The estimated regression coefficient  𝛽1≈-0.05306234  measures how our linear prediction of wage changes if we set the gender variable  𝐷  from 0 to 1, holding the controls  𝑊  fixed 